langchain/chains/conversation/base.py

```diff
@@ -0,0 +1,59 @@
+"""Chain that carries on a conversation and calls an LLM."""
+from typing import Any, Dict, List
+
+from pydantic import BaseModel, Extra
+
+from langchain.chains.base import Memory
+from langchain.chains.conversation.prompt import PROMPT
+from langchain.chains.llm import LLMChain
+from langchain.prompts.base import BasePromptTemplate
+
+
+class ConversationBufferMemory(Memory, BaseModel):
+    """Buffer for storing conversation memory."""
+
+    buffer: str = ""
+    dynamic_key: str = "history"  #: :meta private:
+
+    @property
+    def dynamic_keys(self) -> List[str]:
+        """Will always return list of dynamic keys.
+
+        :meta private:
+        """
+        return [self.dynamic_key]
+
+    def _load_dynamic_keys(self, inputs: Dict[str, Any]) -> Dict[str, str]:
+        """Return history buffer."""
+        return {self.dynamic_keys[0]: self.buffer}
+
+    def _save_context(self, inputs: Dict[str, Any], outputs: Dict[str, str]) -> None:
```

Comment:
I don't really love the leakage of logic from the ConversationChain class into this interface (e.g. not easily having self.input_key and self.output_key available makes the indexing hard). ideas for keeping this more separate?